Modules/_posixsubprocess.c

```diff
@@ -405,6 +451,9 @@ child_exec(char *const exec_array[],
            int errpipe_read, int errpipe_write,
            int close_fds, int restore_signals,
            int call_setsid,
+#ifdef VFORK_USABLE
+           const sigset_t *old_sigs,
+#endif
```

Comment:
To avoid `#ifdef` here and in `do_fork_exec()` while still not depending on `sigset_t` unconditionally, I can change the type of this `old_sigs` to `void *` and add a cast when it's actually used. I'm not sure what's worse -- ugly `#ifdef`s or a bit hacky parameter.

Reply:
I tried removing `_Py_NO_INLINE` and I didn't get any `-Wclobbered` (with GCC 9.3.0). But anyway - even if we get the warning with other GCC versions - we can make it `_Py_NO_INLINE` only if `VFORK_USABLE`. Then, we can remove the parameter `#ifdef` here and in `do_fork_exec`, pass `NULL` in the non-`VFORK_USABLE` case, and trust the compiler to inline it all and eliminate the unneeded parameter since both functions are inlinable.

Reply:
Unfortunately, GCC doesn't inline `child_exec()` even I try to force it with `always_inline`.

But those `#ifdef`s in parameter/argument lists seem much uglier to me than a `void *` parameter, and overhead of passing a useless argument in non-`VFORK_USABLE` case should be negligible, so I'm leaning to getting rid of `#ifdef`s here for the next patch version.